File: src/main/java/com/gmail/borlandlp/minigamesdtools/geoip/ipstack/IpstackResponse.java
package com.gmail.borlandlp.minigamesdtools.geoip.ipstack;

import com.gmail.borlandlp.minigamesdtools.geoip.GeoData;
import org.json.simple.JSONObject;

public class IpstackResponse {
    private Errors error;
    private JSONObject response;

    public void setResponse(JSONObject jsonObject) {
        this.response = jsonObject;
    }

    public Errors getError() {
        return error;
    }

    public void setError(Errors error) {
        this.error = error;
    }

    public boolean isSuccessful() {
        return error == null;
    }

    public GeoData getData() {
        return new IpstackResponseParser(this.response).parse();
    }
}

File: src/main/kotlin/com/gmail/borlandlp/minigamesdtools/geoip/ipstack/IpstackResponse.kt
package com.gmail.borlandlp.minigamesdtools.geoip.ipstack

import com.gmail.borlandlp.minigamesdtools.geoip.GeoData
import org.json.simple.JSONObject

class IpstackResponse {
    var error: ErrorResponse? = null
    private var response: JSONObject? = null

    val isSuccessful: Boolean
        get() = error == null

    val data: GeoData?
        get() = if (response != null) IpstackResponseParser(response!!).parse() else null

    fun setResponse(jsonObject: JSONObject) {
        response = jsonObject
    }
}